### ISSUE TITLE
[BUG] adding displayName to table

### ISSUE THREAD

**Issue by YaseenAlMufti** [exceljs/exceljs#1641]
## 🐛 Bug Report

Adding displayName to the properties of addTable will result in a recoverable corrupted Excel file.

After Excel repair tool, the table will not contain the displayName.

Lib version: 4.2.0

## Steps To Reproduce

1. Create workbook
2. Create worksheet
3. Add Table with displayName property in options.

```javascript
const wb = new ExcelJS.Workbook();
const ws = wb.addWorksheet("Test");

ws.addTable({
   name: 'MyTable', 
   displayName: 'My Test Table',
   ref: 'A1', 
   headerRow: true,    
   totalsRow: true, 
   style: { 
        theme: 'TableStyleDark3', 
        showRowStripes: true, 
   },
   columns: [ 
       {
           name: 'Date', 
           totalsRowLabel: 'Totals:',
           filterButton: true
       }, 
      {
           name: 'Amount', 
           totalsRowFunction: 'sum', 
           filterButton: false
      },
   ], 
   rows: [ 
       [new Date('2019-07-20'), 70.10], 
       [new Date('2019-07-21'), 70.60], 
       [new Date('2019-07-22'), 70.10], 
   ],
})
```

## The expected behaviour:

1. displayName chosen for the table would show on a cell above the table.
2. The file is not corrupted.

## Possible solution (optional, but very helpful):

Removing displayName from the options will fix the corrupted file issue. However, we will have to add the name of the Table manually by figuring out in which cell we will need to add it to.

**Comment by YaseenAlMufti** [comment]
As of now, the things I see broken are like follows:

1. Adding displayName => corrupted file.
2. Adding note and table in the same worksheet => corrupted file.

3. (headerRow: false) in properties => corrupted file.
4. Adding a table with empty rows => corrupted file.

All testing is done on Mac Catalina and exceljs v4.2.0.

**Comment by Jigna-Rathod** [comment]
@YaseenAlMufti I do not have displayName in my table still  the corrupt file issue. below is the code
` worksheet.addTable({
      name: 'cc' + 'E10',
      ref: 'E10',
      headerRow: true,
      style: {
        theme: 'TableStyleLight9',
        showRowStripes: false,
      },
      columns: [{name:'integer'},{name:'chars'}],
      rows: [[12.2,'aa'],[23.3,'cc']],
    });`

**Comment by YaseenAlMufti** [comment]
> @YaseenAlMufti I do not have displayName in my table still  the corrupt file issue. below is the code
> ` worksheet.addTable({
>       name: 'cc' + 'E10',
>       ref: 'E10',
>       headerRow: true,
>       style: {
>         theme: 'TableStyleLight9',
>         showRowStripes: false,
>       },
>       columns: [{name:'integer'},{name:'chars'}],
>       rows: [[12.2,'aa'],[23.3,'cc']],
>     });`

Yeah, it looks like you cannot have any numbers on the "name" property even if its a string.

Try the same code but with only 'cc' or 'cc' + 'E'.
It should work.

**Comment by Jigna-Rathod** [comment]
@YaseenAlMufti It worked. Super thanks.
